anndata/_core/sparse_dataset.py

```diff
@@ -441,20 +452,26 @@ def append(self, sparse_matrix: ss.spmatrix):
         indices.resize((orig_data_size + sparse_matrix.indices.shape[0],))
         indices[orig_data_size:] = sparse_matrix.indices
 
+    @cached_property
+    def indptr(self):
+        arr = self.group["indptr"][...]
+        arr.flags.writeable = False
```

Comment:
I think this should definitely be left in.  There's no reason not to check this.  The only downside is if the underlying array changes in the future to be of a different class, but if I remember correctly, doing subset-write in Torch (so presumably other GPU libs as well) is a huge no-no.

Reply:
In that case we'll have to remove the flag check.

Reply:
I think we can just leave this as writeable since we copy it anyway.